src/locking/LockingManager.ts

```diff
@@ -41,63 +41,16 @@ export default class LockingManager {
         message: `Acquired successfully the lock entity '${lock.entity}' ('${lock.key}') of type '${lock.type}'`,
         detailedMessages: { lock }
       });
-      Utils.isDevelopmentEnv() && console.debug(chalk.green(`Acquire the lock entity '${lock.entity}' ('${lock.key}') of type '${lock.type}' in Tenant ID '${lock.tenantID}'`));
+      Utils.isDevelopmentEnv() && console.debug(chalk.green(`Acquired successfully the lock entity '${lock.entity}' ('${lock.key}') of type '${lock.type}' in Tenant ID '${lock.tenantID}'`));
       return true;
     } catch (error) {
-      await Logging.logWarning({
-        tenantID: lock.tenantID,
-        module: MODULE_NAME, method: 'acquire',
-        action: ServerAction.LOCKING,
-        message: `Cannot acquire the lock entity '${lock.entity}' ('${lock.key}') of type '${lock.type}' in Tenant ID ${lock.tenantID}`,
-        detailedMessages: { lock, error: error.message, stack: error.stack }
-      });
-      Utils.isDevelopmentEnv() && console.error(chalk.red(`Cannot acquire the lock entity '${lock.entity}' ('${lock.key}') of type '${lock.type}' in Tenant ID '${lock.tenantID}'`));
-      return false;
-    }
-  }
-
-  public static async tryAcquire(lock: Lock, timeout: number): Promise<boolean> {
-    let timeoutReached = false;
-    setTimeout(() => {
-      timeoutReached = true;
-    }, timeout);
-    try {
-      switch (lock.type) {
-        case LockType.EXCLUSIVE:
-          // Busy loop tries
-          while (!timeoutReached) {
-            try {
-              await LockingStorage.insertLock(lock);
-              break;
-            } catch {
-              await Utils.sleep(1000);
-            }
-          }
-          if (timeoutReached) {
-            throw Error(`Lock acquisition timeout ${timeout}ms reached`);
-          }
-          break;
-        default:
-          throw new BackendError({
-            action: ServerAction.LOCKING,
-            module: MODULE_NAME, method: 'tryAcquire',
-            message: `Cannot acquire a lock entity '${lock.entity}' ('${lock.key}') with an unknown type '${lock.type}'`,
-            detailedMessages: { lock }
-          });
+      // Check if specific lock for the asset has an expiration date
+      if (retry && await LockingManager.checkAndReleaseExpiredLock(lock)) {
```

Comment:
In my original PR I also checked if the new lock has an expiration date, because normally if the new one does not have an expiration date, the one in the db has none either. 
(So we could reduce data base traffic of the lock collection)
But it should be fine like this also.

Reply:
> In my original PR I also checked if the new lock has an expiration date, because normally if the new one does not have an expiration date, the one in the db has none either.
> (So we could reduce data base traffic of the lock collection)
> But it should be fine like this also.

I wouldn't do this assumption :-).
What matters is if the lock in the DB has an expiration date to remove it.
No correlation between the two.
Btw, do we have to put in place the Job?

Reply:
Okay. 
Normally the job is not needed if we put an expiration date on every lock type. 
If we want to have locks without an expiration date, we would have to discuss when the locks should be released by the job.

Reply:
> Okay.
> Normally the job is not needed if we put an expiration date on every lock type.
> If we want to have locks without an expiration date, we would have to discuss when the locks should be released by the job.

No, it's just to remove expired locks if you do not request them anymore and then have a clean 'locks' collection, but that's okay for the time being :-).

Reply:
Not allowing lock without cleanup once the expiration date is reached is risky for some OCPI related tasks.

Reply:
> Not allowing lock without cleanup once the expiration date is reached is risky for some OCPI related tasks.

It is released when requested again, I don't see the problem here!

Reply:
It will allow to run the same heavy task twice, which defeats all the locking primary purpose. 
For the pull tokens task, it really might happen if the scheduled task interval is higher than the default lock expiration date.

Reply:
> It will allow to run the same heavy task twice, which defeats all the locking primary purpose.
> For the pull tokens task, it really might happen if the scheduled task interval is higher than the default lock expiration date.

I still don't get it why it's a problem!
Releasing an expired lock is an exceptional use case that should never happen.

Reply:
Locking per design are meant to avoid concurrent execution. The introduced workaround to remove stalled locks must be used with caution when use to protect against concurrent execution of long running and heavy process.  
Since it's a workaround but a one that defeats the primary locking design, it probably should not be enabled by default at a low level and only inside the locking helper to avoid human error and ensure the developer build the helper correctly for its target.

Reply:
It's ok as it is.